src/plugins/intel_cpu/src/emitters/jit_snippets_emitters.hpp

```diff
@@ -30,11 +32,28 @@ struct jit_snippets_compile_args {
     std::vector<size_t> output_dims = {};
 };
 ///
-/// \brief    Kernel is the only entry point to Codogen Jit compilation. Kernel calculates appropriate data offsets,
-/// and invokes enclosed outer Tiles. Only 2d Tiles are currently supported, so the emitters should
-/// be organized in the following way:
-/// KernelEmitter {          /* entry point */
-///     TileEmitter {        /* outer tile */
+/// \brief jit_container_emitter designed to wrap Emitters that contain other Emitters (presently KernelEmitter,
+/// TileSchedulerEmitter and TileEmitter). This is needed to provide common interface for register mapping
+/// (abstract to physical) and nested code access.
+///
+class jit_container_emitter: public jit_emitter {
+public:
+    jit_container_emitter(dnnl::impl::cpu::x64::jit_generator* h, dnnl::impl::cpu::x64::cpu_isa_t isa,
+                          const std::shared_ptr<ov::Node>& n);
+    std::vector<AllocatedEmitter> get_nested_code();
```

Comment:
Don't like this method:
1. It returns copy of the vector, not a reference that seems to trigger unnecessary copies. 
2. In most cases body is accessed directly given that it is placed in protected section.

And in general I am not sure we need `jit_container_emitter` wrapper at all. Can leave as is but seems need to refactored with nested Tiles introduction.

Reply:
1, 2 agree, fixed. Thanks for pointing out.
we need `jit_container_emitter` to provide unified interface for register mapping. However, I agree that it'll become redundant if we remove TileScheduler and only Kernel and Tile will have body operation. Note that we might need  a similar interface again (probably provided by Tile base class) if we introduce several Tile types (TileLoop, TileMatrix etc).

Reply:
Speaking of `get_nested_code()`, we indeed use it only for the Tile, so lets move `get_nested_code()` from `jit_container_emitter` to `Tile`